dotnet/src/Connectors/Connectors.OpenAI/OpenAIServiceCollectionExtensions.cs

```diff
@@ -347,7 +348,8 @@ public static class OpenAIServiceCollectionExtensions
         string apiKey,
         string? serviceId = null,
         string? modelId = null,
-        HttpClient? httpClient = null)
+        HttpClient? httpClient = null,
+        int? dimensions = null)
```

Comment:
Should dimensions be earlier in the argument list, e.g. before serviceId?

And do we think there might be other options in the future? Wondering if we should be considering some kind of options bag type.

Reply:
> Should dimensions be earlier in the argument list, e.g. before serviceId?

Yes, and I like this approach better, but adding it as last parameter is less painful in terms of breaking changes, especially if users don't use named parameters.

> And do we think there might be other options in the future? Wondering if we should be considering some kind of options bag type.

With increasing number of arguments in these methods it would be better to rewrite these signatures and replace parameters with config models. In this case, it would be possible to easily add new parameters in the future, support optional parameters and don't care about parameters ordering. But I'm not sure if we want to do such change right now, and again, such refactoring will be breaking change.

We can use hybrid approach, leave parameters as they are today and add new model which will contain only new parameters, but then it will feel strange to pass big list of parameters and still initialize config class at the end.

Reply:
> We can use hybrid approach, leave parameters as they are today and add new model which will contain only new parameters, but then it will feel strange to pass big list of parameters and still initialize config class at the end.

I like this option. 

Might consider [Obsolete] the parameterized one in favor of Options.

Reply:
Added this to the backlog https://github.com/microsoft/semantic-kernel/issues/6081